lang/cs/Org.Apache.REEF.Driver.Tests/TestTaskStates.cs

```diff
@@ -0,0 +1,152 @@
+            Assert.False(taskState.IsFinalState(), "TaskWaitingForClose is not final state."); 
+            Assert.True(taskState.MoveNext(TaskEvent.ClosedTask).Equals(TaskTransitionState.TaskClosedByDriver), "Fail to move to TaskClosedByDriver state.");
+            Assert.True(taskState.IsFinalState(), "TaskClosedByDriver should be a final state.");
+        }
+
+        [Fact]
+        public void TestRunningToFailByEvaluator()
+        {
+            var taskState = new DriverTaskState();
+            taskState.MoveNext(TaskEvent.SubmittedTask);
+            taskState.MoveNext(TaskEvent.RunningTask);
+            Assert.True(taskState.MoveNext(TaskEvent.FailedTaskEvaluatorError).Equals(TaskTransitionState.TaskFailedByEvaluatorFailure), "Fail to move to TaskFailedByEvaluatorFailure state.");
+            Assert.True(taskState.IsFinalState(), "TaskFailedByEvaluatorFailure should be a final state.");
+        }
+
+        [Fact]
+        public void TestRunningToFailByCommuThenEvaluator()
+        {
+            var taskState = new DriverTaskState();
+            taskState.MoveNext(TaskEvent.SubmittedTask);
+            taskState.MoveNext(TaskEvent.RunningTask);
+            Assert.True(taskState.MoveNext(TaskEvent.FailedTaskCommunicationError).Equals(TaskTransitionState.TaskFailedByGroupCommunication), "Fail to move to TaskFailedByGroupCommunication state.");
+            Assert.True(taskState.IsFinalState(), "TaskFailedByGroupCommunication should be a final state.");
+            Assert.True(taskState.MoveNext(TaskEvent.FailedTaskEvaluatorError).Equals(TaskTransitionState.TaskFailedByEvaluatorFailure), "Fail to move to TaskFailedByEvaluatorFailure state.");
+            Assert.True(taskState.IsFinalState(), "TaskFailedByEvaluatorFailure should be a final state.");
+        }
+
+        [Fact]
+        public void TestFromNewToNotAllowed()
+        {
+            var taskState = new DriverTaskState();
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskNew));
+            
+            Action moveNext = () => taskState.MoveNext(TaskEvent.RunningTask);
+            Assert.Throws<ApplicationException>(moveNext);
+
+            moveNext = () => taskState.MoveNext(TaskEvent.WaitingTaskToClose);
+            Assert.Throws<ApplicationException>(moveNext);
+
+            moveNext = () => taskState.MoveNext(TaskEvent.CompletedTask);
+            Assert.Throws<ApplicationException>(moveNext);
+        }
+
+        [Fact]
+        public void TestFromRunningToNotAllowed()
+        {
+            var taskState = new DriverTaskState();
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskNew), "Task initial state is not TaskNew.");
+
+            taskState.MoveNext(TaskEvent.SubmittedTask);
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskSubmitting), "Fail to move to TaskSubmitting state.");
```

Comment:
This case is covered in TestNewToCompleteSenario test, don't need assert here (and the next assert also)